### DISCUSSION TITLE
STM32F1xx  C axis signals

### DISCUSSION THREAD

**Discussion by samitray** [grblHAL/core#502] [General]
Using STM32F1xx build with modified generic_map.h, config.h for N_AXIS = 6,  The first 5 axis are on port A, and C axis step dir is on Port C (14 , 15). Strange behaviour on jogging C axis dir signals are showing up on Port A 15..... all other axis signals are working perfectly. Triple checked the generic_map file, schematics and continuity on PCB,  cant seem to find any issues there, greped M5_STEP_P* , C_STEP_P* and added compile time message to check if port and pins are getting assigned correctly.. all seems ok. Checked soldering and signal lines isolation....all ok... map and config file is attached as txt
[config.txt](https://github.com/grblHAL/core/files/15132059/config.txt)
[generic_map.txt](https://github.com/grblHAL/core/files/15132062/generic_map.txt)

Plan to burn simple blink on PC14,15 and PA14,15 and conduct some sanity checks in the next few days and report back....
.......

**Comment by terjeio** [comment]
`#define STEP_OUTMODE            GPIO_MAP`
requires all step pins to be on the same port, use `GPIO_BITBAND` instead of  `GPIO_MAP`.

**Comment by samitray** [comment]
Yes GPIO_BITBAND works...... 
GPIO_MAP,   GPIO_BITBAND , GPIO_SHIFT stuff is wicked....I remember somewhere in grblMega or smoothie, had a hard time getting all the lines on the same port....... my brain seems to be going bad like my eyesight...
Thanks @terjeio
